### PR TITLE
React settings

### PR DESCRIPTION
**Pull request by el-mapache**
**Should be merged after #493**

* Fixes bugs/adds missing functionality back into site settings page
* Adds basic development logger to the server to let us gain insights into the routes being called
* Adds middleware to call the History singleton from store dispatches

### CODE REVIEW

File: assets/app/actions/siteActions.js [hunk 1]
```diff
@@ -34,13 +34,12 @@ export default {
...
   deleteSite(siteId) {
-    federalist.deleteSite(siteId);
-  },
-
-  deletedSite(siteId) {
-    store.dispatch({
-      type: siteActionTypes.SITE_DELETED,
-      siteId
+    federalist.deleteSite(siteId).then((site) => {
+      console.log('is there a deleted site?', site, arguments);
```
Comment (jeremiak) [inline assets/app/actions/siteActions.js]: @el-mapache should we remove this logging statement?

Comment (el-mapache) [inline assets/app/actions/siteActions.js]: We should, although I don't see it in the code

Comment (jeremiak) [inline assets/app/actions/siteActions.js]: Ya, oops, I was looking at an outdated diff

File: config/http.js [hunk 2]
```diff
@@ -34,13 +34,20 @@ module.exports.http = {
   * router is invoked by the "router" middleware below.)                     *
   *                                                                          *
   ***************************************************************************/
+  requestLogger: function(req, res, next) {
```
Comment (jeremiak) [inline config/http.js]: Woah, this is great!!! We should've done this a long time ago

**Comment by jeremiak** [comment]
Looks good to me!

**Pull request merged**